### PR TITLE
84730

### PR DESCRIPTION
**Pull request by ZASMan**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [84730](https://devops.dchbx.org/redmine/issues/84730) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | Kathryn Sullivan |

### CODE REVIEW

File: app/models/phone.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Phone
   embedded_in :census_member, class_name: "CensusMember"
 
   KINDS = ["home", "work", "mobile", "main", "fax"]
-  OFFICE_KINDS = ["phone main"]
+  OFFICE_KINDS = ["main"]
```
Comment (harshaellanki) [inline app/models/phone.rb]: for employers we are still using   OFFICE_KINDS = ["phone main"], it should not be changed

Comment (ZASMan) [reply app/models/phone.rb]: Great point. I think we need something like this.


```
  validates :kind,
    inclusion: { in: valid_phone_kinds, message: "%{value} is not a valid phone type" },
    allow_blank: false

  def valid_phone_kinds
    if # is a person
      KINDS
    else # is an office
      KINDS + OFFICE_KINDS
    end
  end
```

File: app/views/shared/_phone_fields.html.erb [hunk 2]
```diff
@@ -2,14 +2,16 @@
   <div class="phone col-md-12 col-sm-12 col-xs-12 no-pd">
 
     <%= f.fields_for :phones, errors: {}, fieldset: true do |phone|  %>
+      <% kind = phone.object.kind.upcase %>
+      <% unless kind.downcase == 'phone main' && phone.is_person_phone? %>
```
Comment (ZASMan) [inline app/views/shared/_phone_fields.html.erb]: This is where the underlying fix should take place, preventing them from being created on the form.

File: spec/models/forms/broker_agency_profile_spec.rb [hunk 3]
```diff
@@ -79,7 +79,7 @@
 
   let(:phone_attributes) {
     {
-      kind: "phone main",
+      kind: "main",
```
Comment (ZASMan) [inline spec/models/forms/broker_agency_profile_spec.rb]: This was for a phone for an individual person

**Pull request closed without merging**